src/jsonapi/jsonapi-generator.py

```diff
@@ -237,8 +289,27 @@ def processFile(file):
 			paramsDeclaration = ''
 			for pn in orderedParamNames:
 				mp = paramsMap[pn]
-				paramsDeclaration += '\t\t' + mp._type + ' ' + mp._name
-				if mp._defval != '':
+				# Determine the type for declaration (remove reference qualifier)
+				declaration_type = mp._type
+				is_reference = False
+				if '&' in declaration_type:
+					is_reference = True
+					# Be careful not to remove '&' from pointer types like 'Type *&' if that occurs
+					# Assuming simple references 'Type&' or 'const Type&' for now
+					parts = declaration_type.split('&')
+					if len(parts) == 2 and parts[1].strip() == '': # Check it's likely a simple reference
+						declaration_type = parts[0].strip() 
+					else:
+						# Handle more complex cases or leave as is if unsure
+						# For now, just log a warning if we encounter an unexpected reference type
+						print(f"Warning: Encountered potentially complex reference type '{mp._type}' for parameter '{mp._name}'. Declaration might be incorrect.")
+						declaration_type = declaration_type.replace('&', '').strip() # Fallback: simple removal
```

Comment:
AFAIR if something unexpected/unhanded is encountered compilation must fail, so developers notice ASAP they screw-up something.